adr/001-hosting.md

```diff
@@ -0,0 +1,93 @@
+# Hosting
+
+* Status: proposed
+* Deciders: [Shawn Allen](https://github.com/shawnbot), [HuiRan Shao](https://github.com/hshaosf)
+* Date: 2022-03-01
+
+## Context and Problem Statement
+
+The archive web server will primarily serve as a [URL redirect] engine for legacy sites. It will need to:
+
+1. Accept HTTP requests from multiple domains (respecting the `Host` header)
+2. Redirect known URLs to their new homes on [sf.gov][] (explicit redirects)
+3. Redirect all other document URLs (HTML, PDF, Word, Excel, etc.) to their archived snapshots
+4. Serve helpful [404] responses to all other URLs
+
+From an engineering standpoint, we also expect to:
+
+* Easily run, develop, and test the server locally
+* Run tests locally, in CI, and in review environments
+* Have alerts that notify us when problems arise
+* Have easily accessible logs to diagnose problems
+* Scale the web server to meet high traffic demands
+
+We need to choose a hosting platform that serves these needs.
+
+## Considered Options
+
+* [Azure functions](#azure-functions)
+* [Heroku](#heroku)
+* [Pantheon](#pantheon)
+
+## Decision Outcome
+
+Chosen option: **Heroku**
+
+### Positive Consequences <!-- optional -->
+
+* Quick and easy setup
+* Full control over HTTP stack
+* Virtually limitless implementation options, from web app frameworks to low-level servers like [nginx]
+* Add-ons provide log retention and alerts
+* Ability to scale horizontally
+
+### Negative Consequences <!-- optional -->
+
+* Will cost more than alternatives
+* Requires basic server management
+
+## Pros and Cons of the Options <!-- optional -->
+
+### Azure functions
+
+[Azure functions] are Microsoft's answer to AWS Lambda and other serverless offerings. They can be used for on-demand compute tasks, but our interest is in [HTTP triggers](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=javascript), which run as responders to HTTP requests.
+
+* Good, because we have access to Azure through DT
+* Good, because Microsoft offers great enterprise support
+* Good, because [Application Insights](https://docs.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview) might serve our monitoring/alert needs out of the box
+* Good, because [function proxies](https://docs.microsoft.com/en-us/azure/azure-functions/functions-proxies) may offer a nice management UI for redirect rules
```

Comment:
```suggestion
* Good, because [function proxies](https://docs.microsoft.com/en-us/azure/azure-functions/functions-proxies) may offer a nice management UI for redirect rules
* Good, because functions [scale automatically](https://docs.microsoft.com/en-us/azure/azure-functions/functions-scale)
```